### ISSUE TITLE
Series.filter() chokes on boolean literal

### ISSUE THREAD

**Issue by s-banach** [pola-rs/polars#7205]
### Polars version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Issue description

I was looking at filtering with boolean arrays while thinking about #7200, and I ran into the following error.

### Reproducible example
The following uses `pl.lit(True)` for simplicity, but you get the same issue with `pl.lit(np boolean array)`.
```python
import polars as pl

series = pl.Series(values=[1, 2])
predicate = pl.lit(True)
series.filter(predicate)
"AttributeError: 'Expr' object has no attribute '_s'"
```

### Expected behavior

I expected `Series.filter(pl.lit(True))` to work the same as `DataFrame.filter(pl.lit(True))`.

### Installed versions

<details>

```
---Version info---
Polars: 0.16.9
Index type: UInt32
Platform: Windows-10-10.0.19044-SP0
Python: 3.10.8 | packaged by conda-forge | (main, Nov 24 2022, 14:07:00) [MSC v.1916 64 bit (AMD64)]
---Optional dependencies---
pyarrow: 11.0.0
pandas: 1.5.2
numpy: 1.22.3
fsspec: <not installed>
connectorx: <not installed>
xlsx2csv: <not installed>
deltalake: <not installed>
matplotlib: <not installed>
```

</details>

**Comment by s-banach** [comment]
P.S. Looking at the type annotations, it appears that the Expr `pl.lit(...)` is not even intended to be supported.
Feel free to close this if it's not a bug.

**Comment by ritchie46** [comment]
`Series` cannot accept expression as they lack a context. Where would `col("foo")` search for "foo".

**Comment by s-banach** [comment]
I think the reason I got confused and made this dumb mistake is partly due to the error message from `pli.expr_to_lit_or_expr` that is produced by the following code:
```python
import numpy as np
import polars as pl

c = np.array([True, False])
x = pl.Series("x", [1, 2])
pl.DataFrame([x]).lazy().filter(c).collect()
```
`TypeError: did not expect value [ True False] of type <class 'numpy.ndarray'>, maybe disambiguate with pl.lit or pl.col`

I saw that when I was learning polars a few months ago, and my takeaway was "`pl.lit()`  is a magical incantation to wrap around numpy arrays to make error messages go away". I guess the more natural thing to wrap around numpy arrays would be `pl.Series()`.

I'm not sure if it's possible to give a specific error message about numpy arrays in the function `pli.expr_to_lit_or_expr` that suggests turning them into a Series, of if there would be too much `if NUMPY_IS_AVAILABLE` checking that has to go on and bloat the code.
